### PR TITLE
Dev 2001 timeout log

### PR DESCRIPTION
**Pull request by khaledkbadr**
The timeout log was due to a bug that was fixed in 1.18.1 regarding handling reconnecting 

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md#bug-or-regression-6

### CODE REVIEW

File: kuber/utils.go [hunk 1]
```diff
@@ -40,6 +42,8 @@ func (s *ParentsStore) SetParents(namespace string, kind string, name string, pa
 }
 
 func (s *ParentsStore) GetParents(namespace string, kind string, name string) (*ParentController, bool) {
+	s.Lock()
+	defer s.Unlock()
 	parents, found := s.parents[GetEntityKey(namespace, kind, name)]
```
Comment (iahvector) [inline kuber/utils.go:47]: Why lock on reading?

Comment (khaledkbadr) [reply kuber/utils.go:47]: It appeared as a data race when I ran the build with `-race` flag as if you're going to read a variable that might be set from another goroutine, then you need to lock the associated mutex before you read the value to ensure that you actually see the updates

Comment (khaledkbadr) [reply kuber/utils.go:47]: but I think you're right since all the writes to that map have locks on them. I wonder what triggered the data race though

Comment (khaledkbadr) [reply kuber/utils.go]: I removed it and will inspect the data race when/if it showed up again

File: client/connect.go [hunk 2]
```diff
@@ -95,13 +98,17 @@ func (client *Client) StartWatchdog() {
...
-			if startTime.Add(10 * time.Minute).Before(time.Now()) {
+		client.blockedM.Lock()
+		{
+			if (client.lastSent == time.Time{}) {
+				if startTime.Add(10 * time.Minute).Before(time.Now()) {
+					break
+				}
+			} else if client.lastSent.Add(10 * time.Minute).Before(time.Now()) {
 				break
 			}
```
Comment (iahvector) [inline client/connect.go:109]: This here is another locked read. I think you can remove the lock and/or extract `lastSent` in a variable so it's read only once per iteration.

Comment (khaledkbadr) [reply client/connect.go:109]: there're writes below in line 107  `client.lastSent.Add(10 * time.Minute).Before(time.Now())`

**Pull request merged**